### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by paulinagonzalezc**
## OOP school library: set up associations 🌟

In this project, I finished creating the remaining classes for my school library and created the associations between them

* Create a class Classroom
* Create the has-many/belongs-to relationship between Classroom and Student
* Create a class Book
* Create a class Rental
* Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (m05e5) [inline book.rb]: - [x] Hi. as part of the requirement, you are supposed to create a has-many relationship side of Book and Rental (a book has many rentals). So, you are supposed to create a method for adding rentals in the book class and the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association.

File: person.rb [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
+  attr_reader :id, :rentals
+
   def initialize(age, name = 'Unknown', parent_permission: true)
     @id = Random.rand(1..100)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
     super()
   end
```
Comment (m05e5) [inline person.rb:16]: - [x] Hi. as part of the requirement, you are supposed to create a has-many relationship side of Person and Rental (a person has many rentals). So, you are supposed to create a method for adding rentals in the person class and the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association.

**Review by m05e5 — CHANGES_REQUESTED** [review]
Hi @paulinagonzalezc , 

Good job so far. !
There are some issues that you still need to work on to go to the next project but you are almost there!
mario_ladders


## Highlights

- Good PR description
- Clean and well-organized code


## Required Changes ♻️
* You really did a great job but there are some features you need to improve

- [ ] Please check the comments below 

_I'm sure you can do it just keep it up_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_@m05e5_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (paulinagonzalezc) [reply book.rb]: Thank you @m05e5 , let me add that part!

File: person.rb (hunk 2 above)
Comment (paulinagonzalezc) [reply person.rb:16]: Thank you @m05e5 , let me add that too!

**Review by DammyShittu — APPROVED** [review]
Hi @paulinagonzalezc,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**